src/clarity-icons/clarity-icons-element.ts

```diff
@@ -55,25 +59,99 @@ const setIconSize = function(element: any, size: string) {
 };
 
 ClarityIconElement.prototype.connectedCallback = function() {
+    // One thing to note here is that the attributeChangedCallback method is called for every attribute first
+    // before this connectedCallback method called only once when the custom element is inserted into the DOM.
+    // So we could check whether the attribute values really changed or not.
+    // If not, we don't need to execute the same codes again.
     const host = this as HTMLElement;
 
     if (host.hasAttribute("shape")) {
-        generateIcon(host, host.getAttribute("shape"));
+        const shapeAttrValue = host.getAttribute("shape").split(/\s/)[0];
+
+        if (shapeAttrValue === this.currentShapeAttrVal) {
```

Comment:
You inlined this from the function, but this test makes no sense for the `connectedCallback`, `this.currentShapeAttrVal` will always be undefined. It's only here for when the shape attribute changes.

Reply:
We set value to `this.currentShapeAttrVal` in `attributeChangedCallback()`, which gets invoked before `connectedCallback()`. So it won't always be undefined.

Reply:
I do think there are cases that users may want to manipulate `clr-icon` elements before or without injecting them into the DOM tree. In those cases, `clr-icon` element's attributes should update properly. Also, in environments with simulated or fake DOM tree, clr-icon attributes probably wouldn't work. For example, if I do:
```
let myIcon = document.createElement("clr-icon");
myIcon.setAttribute("shape", "user");
console.log(myIcon.innerHTML); // no SVG template as we have to now append to the DOM.
```